src/Domain/CoordinateMaps/TimeDependent/Translation.cpp

```diff
@@ -96,42 +165,175 @@ std::array<tt::remove_cvref_wrap_t<T>, Dim> Translation<Dim>::frame_velocity(
+  } else {
+    return determinant_and_inverse(
+               jacobian(source_coords, time, functions_of_time))
+        .second;
+  }
 }
 
 template <size_t Dim>
-template <typename T>
-tnsr::Ij<tt::remove_cvref_wrap_t<T>, Dim, Frame::NoFrame>
-Translation<Dim>::inv_jacobian(const std::array<T, Dim>& source_coords) const {
-  return identity<Dim>(dereference_wrapper(source_coords[0]));
+double Translation<Dim>::root_finder(const double& translated_radius,
+                                     const double& magnitude_function_of_time,
+                                     const double& root_factor) const {
+  const double center_offset = sqrt(magnitude_function_of_time);
+  double lower_bound = (translated_radius - center_offset) * (1.0 - 1.e-9);
+  double upper_bound = (translated_radius + center_offset) * (1.0 + 1.e-9);
+  if (lower_bound < 0.) {
+    lower_bound = 0.;
+  }
+  // If the upper and lower bounds are < 0., then there's no need to find a root
+  // the root will be 0.
+  if (upper_bound < 0.) {
+    return 0.;
+  }
+  double function_at_lower_bound = std::numeric_limits<double>::min();
+  double function_at_upper_bound = (*f_of_r_)(upper_bound);
+  // bracket function requires a function to return std::optionals
+  const auto radius_bracket_function =
+      [this, &translated_radius, &magnitude_function_of_time,
+       &root_factor](const double x) -> std::optional<double> {
+    return square(translated_radius) +
+           ((*f_of_r_)(x)) *
+               (root_factor + magnitude_function_of_time * (*f_of_r_)(x)) -
+           square(x);
+  };
+  RootFinder::bracket_possibly_undefined_function_in_interval(
+      &lower_bound, &upper_bound, &function_at_lower_bound,
+      &function_at_upper_bound, radius_bracket_function);
+  const double absolute_tol = std::numeric_limits<double>::epsilon() *
+                              std::max(translated_radius, center_offset);
+  const double relative_tol = 2.0 * std::numeric_limits<double>::epsilon();
+  const auto wrapper = [this, &translated_radius, &magnitude_function_of_time,
+                        &root_factor](const double x) {
+    return square(translated_radius) +
+           ((*f_of_r_)(x)) *
+               (root_factor + magnitude_function_of_time * (*f_of_r_)(x)) -
+           square(x);
+  };
+  return RootFinder::toms748(wrapper, lower_bound, upper_bound,
+                             function_at_lower_bound, function_at_upper_bound,
+                             absolute_tol, relative_tol);
+  return 0.;
 }
 
 template <size_t Dim>
 void Translation<Dim>::pup(PUP::er& p) {
-  size_t version = 0;
+  size_t version = 1;
   p | version;
   // Remember to increment the version number when making changes to this
   // function. Retain support for unpacking data written by previous versions
   // whenever possible. See `Domain` docs for details.
   if (version >= 0) {
     p | f_of_t_name_;
   }
+  if (version >= 1) {
+    p | f_of_r_;
+    p | center_;
+  }
```

Comment:
You also need an `else if (p.isUnpacking())` clause here. This is because if we are reading a v0 object, we still need to set `f_of_r_` and `center_` to something because this is a v1 object. So you should pick defaults that match what a v0 object would expect (here, namely `nullptr` and 0).